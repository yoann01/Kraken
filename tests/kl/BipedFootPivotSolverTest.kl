
require Kraken;

operator entry(){
  BipedFootPivotSolver solver();

  Boolean drawDebug = true;
  Scalar rigScale = 1.0;

  Boolean rightSide = false;
  Scalar footRock = 0.0;
  Scalar footBank = 0.0;
  Mat44 pivotAll = Xfo(Quat(Vec3(0.0,0.0,0.0),1.0), Vec3(1.75,1.14999997616,-1.25), Vec3(1.0,1.0,1.0)).toMat44();
  Mat44 backPivot = Xfo(Quat(Vec3(0.0,0.0,0.0),1.0), Vec3(1.75,0.0,-2.5), Vec3(1.0,1.0,1.0)).toMat44();
  Mat44 frontPivot = Xfo(Quat(Vec3(0.0,0.0,0.0),1.0), Vec3(1.75,0.0,2.0), Vec3(1.0,1.0,1.0)).toMat44();
  Mat44 outerPivot = Xfo(Quat(Vec3(0.0,0.0,0.0),1.0), Vec3(2.5,0.0,-1.25), Vec3(1.0,1.0,1.0)).toMat44();
  Mat44 innerPivot = Xfo(Quat(Vec3(0.0,0.0,0.0),1.0), Vec3(1.0,0.0,-1.25), Vec3(1.0,1.0,1.0)).toMat44();

  Mat44 result;

  solver.solve(drawDebug, rigScale, rightSide, footRock, footBank, pivotAll, backPivot, frontPivot, outerPivot, innerPivot, result);

  Xfo pivot_result = Xfo(result);
  report('Pivot Xfo: ' + pivot_result);
}
