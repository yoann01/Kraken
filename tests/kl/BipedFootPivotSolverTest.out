[FABRIC:MT] Ignored extension directory 'D:\fabric\FabricEngine-2.1.0-Windows-x86_64\Tests\Exts': unable to open directory 'D:\fabric\FabricEngine-2.1.0-Windows-x86_64\Tests\Exts': The system cannot find the path specified. (code 3 [0x3])Pivot Xfo: {ori:{v:{x:+0.0,y:+0.0,z:+0.0},w:+1.0},tr:{x:+1.75,y:+1.15,z:-1.25},sc:{x:+1.0,y:+1.0,z:+1.0}}